MoneyDataType/Currency.extra.cs

```diff
@@ -1,174 +1,176 @@
+    public static ICurrency UzbekistaniSom => _defaultProvider.GetCurrency("UZS");
+    public static ICurrency VenezuelanBolívar => _defaultProvider.GetCurrency("VES");
+    public static ICurrency VietnameseDong => _defaultProvider.GetCurrency("VND");
+    public static ICurrency CentralAfricanCfaFranc => _defaultProvider.GetCurrency("XAF");
+    public static ICurrency EastCaribbeanDollar => _defaultProvider.GetCurrency("XCD");
+    public static ICurrency SpecialDrawingRights => _defaultProvider.GetCurrency("XDR");
+    public static ICurrency WestAfricanCfaFranc => _defaultProvider.GetCurrency("XOF");
+    public static ICurrency YemeniRial => _defaultProvider.GetCurrency("YER");
+    public static ICurrency SouthAfricanRand => _defaultProvider.GetCurrency("ZAR");
 
-        private static readonly ICurrencyProvider DefaultProvider = new CurrencyProvider();
+    private static readonly ICurrencyProvider _defaultProvider = new CurrencyProvider();
 
-        public static ICurrency FromISOCode(string code, IEnumerable<ICurrencyProvider> providers = null)
-        {
-            var found = DefaultProvider.GetCurrency(code);
-            return found ?? (providers ?? new List<ICurrencyProvider>())
-                .SelectMany(p => p.Currencies)
-                .FirstOrDefault(c => c.CurrencyIsoCode == code);
-        }
+    public static ICurrency FromIsoCode(string code, IEnumerable<ICurrencyProvider> providers = null)
+    {
+        var found = _defaultProvider.GetCurrency(code);
+        return found ?? (providers ?? new List<ICurrencyProvider>())
+            .SelectMany(p => p.Currencies)
+            .FirstOrDefault(c => c.CurrencyIsoCode == code);
+    }
 
-        public static ICurrency FromNativeName(string name, IEnumerable<ICurrencyProvider> providers = null)
+    public static ICurrency FromNativeName(string name, IEnumerable<ICurrencyProvider> providers = null)
+    {
+        if (string.IsNullOrWhiteSpace(name))
         {
-            if (string.IsNullOrWhiteSpace(name))
-                throw new System.ArgumentException("Must provide a name", nameof(name));
-            return (providers ?? new List<ICurrencyProvider>()).SelectMany(p => p.Currencies).FirstOrDefault(c => c.NativeName == name);
+            throw new ArgumentException("Must provide a name", nameof(name));
         }
 
-        public static ICurrency FromEnglishName(string name, IEnumerable<ICurrencyProvider> providers = null)
-        {
-            if (string.IsNullOrWhiteSpace(name))
-                throw new System.ArgumentException("Must provide a name", nameof(name));
-            return (providers ?? new List<ICurrencyProvider>()).SelectMany(p => p.Currencies).FirstOrDefault(c => c.EnglishName == name);
-        }
+        return (providers ?? new List<ICurrencyProvider>())
+            .SelectMany(p => p.Currencies)
+            .FirstOrDefault(c => c.NativeName == name);
```

Comment:
I'm annoyed that everyone wants to put the`?` is on the next line together with the dot, instead of `providers?` which is syntactically still fine and semantically better. I realize I'm in the minority with this opinion, just venting...

Reply:
I don't like how it looks, either, but I'd be afraid that one might not spot the `?` at the end of the previous line easily. And also, if we put operators on new lines, let's do it with all of them, and not just halves.

Reply:
Why is it important to spot the `?` easily?

Reply:
I think it's mostly about splitting the operator - that feels weirder than the broken alignment with the `.` below.

Reply:
I think the operator is just that combination for some technical reasons (probably the same reason why we don't have `?()` operator). We have `?[]` and `?.` and in both cases it is a null-conditional plus `[]` or `.` respectively. Also if MS was serious about it being one atomic operator then the compiler wouldn't let us split it. At least that's why I don't think it's weird.